src/input.rs

```diff
@@ -279,14 +304,18 @@ mod test {
 
     #[test]
     fn test_backspace() {
-        line_match("this is the\x7f first\x7f\x7f test", Some("this is th fir test"));
-        line_match("this is the seco\x7fnd test\x7f", Some("this is the secnd tes"));
+        line_match("this is the\x7f first\x7f\x7f test",
```

Comment:
Most of those changes are cause by rustfmt ; better settings could possibly reduce the diff.